eWallet/user/views.py

```diff
@@ -134,6 +135,22 @@ def get(self,request):
         user=request.user
         return render(request, 'StatementHistory.html', {'user': user, 'Signed_in': True})
     
+    def post(self,request):
+        contact=request.POST.get('contact')
+        
+        if User.objects.filter(contact=contact).exists():
+            user=User.objects.get(contact=contact)
+            transfers=Transaction.objects.filter(user=user)
+            heading='Transaction History of '+transfers.first().user.username
+            return render(request,'accountStatements.html',{'user':request.user,'Signed_in':True,'transfers':transfers,'heading':heading})
```

Comment:
why are you passing these as context to the template. request.user is accessible in templates and the heading can be created in templates using filters. Correct this in other places as well.